src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2568,14 +2568,18 @@ private boolean handleSpecialTTEFunctions(
 
     private String getSignatureForInstanceTTEFn(
         String propName, List<String> classTemplateTypeNames, FunctionType ftype) {
-      // If ftype is foo.bar.Promise.prototype.then, extract className as Promise.
-      String className = getUnqualifiedName(getNamespace(getNamespace(ftype.getDisplayName())));
+      // If ftype is foo.bar.Promise.prototype.then, extract className as ಠ_ಠ.clutz.foo.bar.Promise.
+      String className =
+          Constants.INTERNAL_NAMESPACE + "." + getNamespace(getNamespace(ftype.getDisplayName()));
       Iterator<String> templateTypeNames = classTemplateTypeNames.iterator();
       if (!templateTypeNames.hasNext()) return null;
 
       String templateVarName = templateTypeNames.next();
 
-      String classTemplatizedType = className + " < RESULT >";
+      // TODO(lucassloan): goog.Promise has bad types that are coerced to any, so explicitly emit any
```

Comment:
expand 'bad types' -> because of change from 1 type param to 2.

Reply:
Done.